env/ipp_aes_ctr_provider.h

```diff
@@ -0,0 +1,109 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  Copyright (c) 2020 Intel Corporation
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#if !defined(ROCKSDB_LITE)
+
+#if defined(IPPCP) && (defined(HAVE_SSE42) || defined(HAVE_SSE2))
+
+// Includes Intel's Integrated Performance Primitives for Cryptography (IPPCP).
+// IPPCP is lightweight cryptography library that is highly-optimized for
+// various Intel CPUs.
+//
+// We use it here to provide an AES-128/192/256 encryption with a CTR mode of
+// operation.
+//
+// Download URL: https://github.com/intel/ipp-crypto.
+//
+#include <emmintrin.h>
+#include <ippcp.h>
```

Comment:
By using a forward declaration of IppsAESSpec and moving IppAESCTRCipherStream into the C++ file, you could do away with the ifdefs and system includes here.  Might make things a little cleaner...

Reply:
Fixed.